libraries/engage/page/widgets/NestedCategoryFilter/NestedCategoryFilter.jsx

```diff
@@ -0,0 +1,177 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { I18n, ButtonLink } from '@shopgate/engage/components';
+import { bin2hex } from '@shopgate/engage/core/helpers';
+import { CATEGORY_PATH } from '@shopgate/engage/category';
+import { makeStyles } from '@shopgate/engage/styles';
+import { themeConfig } from '@shopgate/engage';
+import { useRoute, usePrevious, useLocalStorage } from '@shopgate/engage/core/hooks';
+import { router } from '@virtuous/conductor';
+import CategoryPicker from './components/Picker';
+import { useNestedCategoryFilterWidget } from './hooks';
+import WidgetHeadline from '../../components/WidgetHeadline';
+
+const { colors, variables } = themeConfig;
+
+const useStyles = makeStyles()({
+  container: {
+    background: colors.light,
+    display: 'flex',
+    flexDirection: 'column',
+    paddingBottom: variables.gap.big,
+  },
+  buttonContainer: {
+    padding: `0 ${variables.gap.big}px`,
+  },
+  button: {
+    width: '100%',
+  },
+});
+
+/**
+ * The NestedCategoryFilter component
+ * @returns {JSX.Element}
+ */
+const NestedCategoryFilter = () => {
+  const { classes } = useStyles();
+  const { state: routeState, id: routeId } = useRoute();
+
+  const {
+    maxDepth = 3,
+    category = '',
+    code,
+    showHeadline,
+    headline,
+    rememberSelection,
+    ...labels
+  } = useNestedCategoryFilterWidget();
+
+  const prevCategory = usePrevious(category);
+
+  const LOCAL_STORAGE_KEY = `nestedCategoryFilterState-${code}`;
+  const [localStorageState, setLocalStorageState] = useLocalStorage(LOCAL_STORAGE_KEY);
```

Comment:
there was an issue with the rehydration of the `buttonCategoryId` state from local storage. since route and local storage persistence is very similar and the PO asked for a testing version, i did some refactoring of the state handling (looks bigger than it is). basically i merged the both states to a single one - like it was in the original implementation.

lets talk about this when you're back.